### DISCUSSION TITLE
Common network topologies?

### DISCUSSION THREAD

**Discussion by boaks** [openthread/openthread#7494] [Q&A] [answered]
Does someone has an overview, what is commonly used with openthread?

Are (coap-)hops (coap2coap proxies) in use?
Protocol translating gateways?
What do you use yourself?

**Comment by jwhui** [comment, marked as answer]
Given that Thread support the Internet Protocol (IP), many application layers utilizing Thread communicate end-to-end and do not require application or protocol gateways, including Matter, HomeKit, and Weave.

At Google I/O 2019, we presented a demo of communicating CoAP end-to-end between Thread devices and cloud end points - see https://www.youtube.com/watch?v=MpkdokSyQIE

That said, Thread does not preclude the use of application/protocol translation gateways. For example, I've seen examples of MQTT-SN used with Thread.

**Comment by boaks** [comment]
Thanks! That reflects also my understanding (and opinion).

A couple of days I was just reading [http://ilab-pub.imp.fu-berlin.de/papers/gasw-cosit-22.pdf](http://ilab-pub.imp.fu-berlin.de/papers/gasw-cosit-22.pdf) and I struggled with the saying "These IoT messages frequently traverse protocol translators at gateways, which break end-to-end transport and security of Internet protocols.". Therefore I tried to get a wider picture.

(I know that video and also [cloud-iot-coap-proxy](https://cloud.google.com/community/tutorials/cloud-iot-coap-proxy).
I also asked to [update](https://github.com/GoogleCloudPlatform/community/issues/2134) it to a more recent version of [eclipse/californium](https://github.com/eclipse/californium). )
